src/controller/motor.c

```diff
@@ -622,15 +606,17 @@ void TIM1_CAP_COM_IRQHandler(void) __interrupt(TIM1_CAP_COM_IRQHANDLER)
   // do not execute all, otherwise ui8_duty_cycle would be decremented more than onece on each PWM cycle
 
   static uint8_t ui8_current_controller_counter;
+  static uint16_t ui16_counter_duty_cycle_ramp_up;
+  static uint16_t ui16_counter_duty_cycle_ramp_down;
   
   // do not control current at every PWM cycle, that will measure and control too fast. Use counter to limit 
   if (++ui8_current_controller_counter > 12)
   {
     // reset counter
     ui8_current_controller_counter = 0;
     
-    // if battery max current or phase current is too much, reduce duty cycle
-    if ((ui8_g_adc_battery_current > ui8_controller_adc_battery_max_current) || (ui8_adc_motor_phase_current > ui8_adc_target_motor_phase_max_current))
+    // if battery max current or motor phase current is too high, reduce duty cycle
+    if ((ui8_g_adc_battery_current > ui8_controller_adc_battery_current_target) || (ui8_adc_motor_phase_current > ui8_adc_target_motor_phase_max_current))
```

Comment:
Are you absolutely certain that you want to remove the check to see if the battery current is more than the maximum allowed?
Did you maybe intend to add a check, instead of replacing one ?

Reply:
Did not remove anything. Also not the max, it is the target.

Reply:
In the old code,  there's a check reading  "battery_current > battery_max_current", presumably to protect against an over-current situation.
In the new code, you're only comparing to current_target, no longer to the maximum.

Reply:
Same thing and same checks. Please look at the code more closely if concerned.